internal/log/log.go

```diff
@@ -38,6 +38,7 @@ var (
 
 // UseLogger sets l as the active logger.
 func UseLogger(l ddtrace.Logger) {
+	Flush()
```

Comment:
Some of the logWriter tests broke other tests because error logs were getting flushed later to a different logger.
When we switch out a logger, we should ensure anything that was written to it is flushed.